### ISSUE TITLE
ArcFace layer works fine on CUDA but worse on Habana

### ISSUE THREAD

**Issue by anti-machinee** [HabanaAI/Model-References#11]
- On CUDA, I am using arcface layer as Fully connected layer in face recognition (input -> resnet -> arcface layer -> loss). Training works fine, loss decreases to 0, accuracy (accuracy each batch) increases. 
- But on Habana:
1. Loss decreases but in the same epoch as CUDA training, loss is about 5 - 9 which converges slowly. I keep training but loss does not decrease to 0
2. Accuracy (average each batch) is always 0
- Here is my source code of arcface layer
![image](https://user-images.githubusercontent.com/81377071/163530983-55be6315-817a-4bda-93d3-5f6bdfded5fb.png)

- I replaces Arcface layer by nn.Linear, it works fine on both CUDA and Habana
Any developer could help me, thank you

**Comment by anti-machinee** [comment]
@greg-serochi please help and support, thank you
Does Habana work if I declare weight like that

**Comment by anti-machinee** [comment]
It works when I deleted `to.(device)` in "self.weight = nn.Parameter(torch.FloatTensor(num_classes, in_size)).to(device)"
Because https://github.com/pytorch/pytorch/issues/17484
